### PR TITLE
Change default install path to ./.bundle/gems [WIP]

### PR DESCRIPTION
**Pull request by smlance**

### CODE REVIEW

File: lib/bundler/settings.rb [hunk 1]
```diff
@@ -138,7 +138,7 @@ def path
       if path = self[:path]
         "#{path}/#{Bundler.ruby_scope}"
       else
-        Bundler.rubygems.gem_dir
+        Pathname.new(File.dirname(__FILE__)).join(".bundle/gems")
```
Comment (segiddins) [inline lib/bundler/settings.rb]: Should this take into account the ruby scope?

Comment (smlance) [inline lib/bundler/settings.rb]: `Bundler.ruby_scope` is a string like `ruby/1.9.1`; why would this be necessary in the path?

Comment (indirect) [inline lib/bundler/settings.rb]: @smlance the Ruby scope string defines the Ruby ABI (application binary interface), which breaks compatibility for compiled gems when it changes. The path gems are installed to always needs to include it.

@segiddins the Rubygems gem path already includes the scope string, so we don't need to add it ourselves.

Comment (segiddins) [inline lib/bundler/settings.rb]: Except this is no longer using the RG path?

-Samuel E. Giddins

On Jun 19, 2015, at 12:46 PM, André Arko <notifications@github.com> wrote:

In lib/bundler/settings.rb:

> @@ -138,7 +138,7 @@ def path
>        if path = self[:path]
>          "#{path}/#{Bundler.ruby_scope}"
>        else
> -        Bundler.rubygems.gem_dir
> +        Pathname.new(File.dirname(__FILE__)).join(".bundle/gems")
@smlance the Ruby scope string defines the Ruby ABI (application binary interface), which breaks compatibility for compiled gems when it changes. The path gems are installed to always needs to include it.

@segiddins the Rubygems gem path already includes the scope string, so we don't need to add it ourselves.

File: lib/bundler/settings.rb [hunk 2]
```diff
@@ -138,7 +138,7 @@ def path
       if path = self[:path]
         "#{path}/#{Bundler.ruby_scope}"
       else
-        Bundler.rubygems.gem_dir
+        Pathname.new(@root)
```
Comment (indirect) [inline lib/bundler/settings.rb]: Ah, this needs to be `File.join(@root, Bundler.ruby_scope)`.

File: spec/support/path.rb [hunk 3]
```diff
@@ -15,7 +15,9 @@ def home(*path)
     end
 
     def default_bundle_path(*path)
-      system_gem_path(*path)
+      root = tmp.join("bundled_app")
+      FileUtils.mkdir_p(root)
+      root.join(".bundle").join(*path)
```
Comment (indirect) [inline spec/support/path.rb]: You can definitely implement this in terms of `bundled_app`, defined below. Something like `bundled_app(".bundle", Bundler.ruby_scope)` should do it.

**Comment by indirect** [comment]
Here's an example of fixing a test by changing the assumption about where gems get installed: https://github.com/bundler/bundler/tree/default-install-path-2-0.

In the future, if you push branches to the bundler/bundler repo for opening pull requests, we can collaborate on them... might be easier. :)

**Comment by indirect** [comment]
Moving this to #3784.

**Pull request closed without merging**